python/taichi/lang/struct.py

```diff
@@ -632,7 +664,41 @@ def filled_with_scalar(self, value):
         return Struct(entries)
 
     def field(self, **kwargs):
-        return Struct.field(self.members, **kwargs)
+        return Struct.field(self.members, self.methods, **kwargs)
+
+
+def struct_class(cls):
```

Comment:
I'm not so sure whether the name `ti.struct_class` is easy enough to understand. Is something like `ti.struct_with_methods` a more straightforward choice? @k-ye

Reply:
Hmm, I actually feel like `struct_class` OK enough. Do you think it's not straightforward from the C++ perspective..?

Reply:
To some extent yes :-) Putting `struct` and `class` together confuses me a bit. But it may be OK if we understand it as "turning a Python class into a Taichi struct".